src/main/java/org/opentestsystem/ss/contentspecid/ContentSpecId.java

```diff
@@ -0,0 +1,220 @@
+package org.opentestsystem.ss.contentspecid;
+
+
+import lombok.Data;
+import org.opentestsystem.ss.contentspecid.enums.ContentSpecClaim;
+import org.opentestsystem.ss.contentspecid.enums.ContentSpecEmphasis;
+import org.opentestsystem.ss.contentspecid.enums.ContentSpecGrade;
+import org.opentestsystem.ss.contentspecid.enums.ContentSpecFormat;
+import org.opentestsystem.ss.contentspecid.enums.ContentSpecSubject;
+import org.opentestsystem.ss.contentspecid.exceptions.IdValidationException;
+import org.opentestsystem.ss.contentspecid.handlers.DomainHandler;
+import org.opentestsystem.ss.contentspecid.handlers.EmphasisHandler;
+import org.opentestsystem.ss.contentspecid.handlers.TargetHandler;
+
+import java.util.Objects;
+
+import static org.apache.commons.lang3.StringUtils.isBlank;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+import static org.opentestsystem.ss.contentspecid.exceptions.ErrorSeverity.INVALID;
+
+/**
+ * Created by Greg Charles on 11/8/18.
+ */
+@Data
+public class ContentSpecId {
```

Comment:
Now that I've read through the entire review it's not obvious the public API one should use.  I'm going to list my understanding of what pieces do to make sure I understand the library:

- ContentSpecId: Knows how to take the parts of an id and create a legacy or enhanced id. Also validates the format
- Handlers: Class that process a part of the ID String and returns the enum value.  For example, EmphasisHandler knows how to convert an emphasis String into an enu
- Converters: knows how parse a String ID (legacy or format)

Is my understanding correct?

I would try to separate the concerns.  You could have the following:
* Parser converts either string into common model
* Validator validates against the model
* Converter knows how to take the common model and transform it into Enhanced or Legacy
* Add a public API and implementation that ties all three together which is the only thing users of the library need to use.